### PR TITLE
Fixes spaces in readline tab completer

### LINKED ISSUE #2485: Tab completion breaks on paths with spaces
```
de3lxl-106789 /tmp  $ xonfig
+------------------+----------------------+
| xonsh            | 0.5.12.dev47.dev102  |
| Git SHA          | c787f9ab             |
| Commit Date      | Aug 16 00:10:57 2017 |
| Python           | 3.4.3                |
| PLY              | 3.8                  |
| have readline    | True                 |
| prompt toolkit   | 1.0.0                |
| shell type       | readline             |
| pygments         | 1.6                  |
| on posix         | True                 |
| on linux         | True                 |
| distro           | Ubuntu               |
| on darwin        | False                |
| on windows       | False                |
| on cygwin        | False                |
| is superuser     | False                |
| default encoding | utf-8                |
| xonsh encoding   | utf-8                |
| encoding errors  | surrogateescape      |
+------------------+----------------------+
de3lxl-106789 /tmp  $ mkdir 'a space'

de3lxl-106789 /tmp  $ touch 'a space/test.txt'
de3lxl-106789 /tmp  $ cat 'a space'<tab>
de3lxl-106789 /tmp  $ cat 'a 'a space/test.txt'
```
The last two lines demonstrate the problem: Pressing <tab> in the second-to-last line at the where indicated produces the last line (note the double 'a), this doesn't seem to be a mere visual/refreshing issue, the produced command is actually not understood by xonsh (pressing return prompts for more input because of the resulting uneven number of ').

OS is Ubuntu 14.04.5 LTS, terminal is rxvt-unicode-256color.

### PR DESCRIPTION
**Pull request by scopatz**
Should fix #2485

### CODE REVIEW

**Comment by gforsyth** [comment]
Thanks, @scopatz!  This fixes the issue reported in #2485 but introduces a new problem

This works now:
```
$ cat 'a space'<TAB>
$ cat 'a space/'<TAB>
$ cat 'a space/test.txt'
```

But this happens, too:
```
$ cat a<TAB>
$ cat space/'
```

**Comment by scopatz** [comment]
Ahhh damn, I knew this was too easy :)

**Comment by scopatz** [comment]
Sorry for the delay getting back to this, but I *think* I have it working now @gforsyth...

**Comment by gforsyth** [comment]
Hey @scopatz -- behavior is much improved and all of my previous cases work, but I just noticed (sorry) that the listing of autocomplete possibilities is incorrectly trimming the front.

```
gil@bad_cat ~/tmp/new 🐚  mkdir 'a space'
gil@bad_cat ~/tmp/new 🐚  touch 'a space/test.txt'
gil@bad_cat ~/tmp/new 🐚  cd 
../      ./       space/'
```

**Comment by scopatz** [comment]
Ahh good catch.  Pushed up another fix, @gforsyth

**Comment by gforsyth** [comment]
Options are now displayed correctly -- but there's either a regression or a new corner-case:

```console
$ ls
'a space'
$ cd 'a<TAB>
$ cd a space/'
```

**Comment by scopatz** [comment]
Thanks again @gforsyth - I think I have put in a fix for this now too.

File: docs/intro.rst [hunk 1]
```diff
@@ -42,6 +42,7 @@ the xonsh shell
         "Conches for the xonsh god!",
         "Python-powered, cross-platform, Unix-gazing shell",
         "Tab completion in Alderaan places",
+        "This fix was trickier expected",
```
Comment (gforsyth) [inline docs/intro.rst]: :heart:

File: xonsh/completers/bash_completion.py [hunk 2]
```diff
@@ -270,7 +270,8 @@ def bash_completions(prefix, line, begidx, endidx, env=None, paths=None,
         from the environment and platform.
     quote_paths : callable, optional
         A functions that quotes file system paths. You shouldn't normally need
-        this as the default is acceptable 99+% of the time.
+        this as the default is acceptable 99+% of the time. This function should
```
Comment (gforsyth) [inline xonsh/completers/bash_completion.py]: There's a few words missing here

File: xonsh/completers/bash_completion.py [hunk 3]
```diff
@@ -335,7 +336,8 @@ def bash_completions(prefix, line, begidx, endidx, env=None, paths=None,
         strip_len += 1
 
     if '-o noquote' not in complete_stmt:
-        out = quote_paths(out, '', '')
+        out, need_qoutes = quote_paths(out, '', '')
```
Comment (gforsyth) [inline xonsh/completers/bash_completion.py]: `need_quotes`

File: xonsh/completers/bash_completion.py [hunk 4]
```diff
@@ -335,7 +336,8 @@ def bash_completions(prefix, line, begidx, endidx, env=None, paths=None,
...
+        strip_len += int(need_qoutes)
```
Comment (gforsyth) [inline xonsh/completers/bash_completion.py]: `need_quotes`

File: xonsh/readline_shell.py [hunk 5]
```diff
@@ -341,9 +382,31 @@ def completedefault(self, prefix, line, begidx, endidx):
...
         show_completions = self._querycompletions(completions, endidx - begidx)
-        return rtn_completions if show_completions else []
+        if show_completions == 0:
+            return []
+        elif show_completions == 1:
+            return rtn
+        elif show_completions == 2:
+            return completions
+        else:
+            raise ValueError('query compeltions flag not understood.')
```
Comment (gforsyth) [inline xonsh/readline_shell.py]: *completions

File: xonsh/xonfig.py [hunk 6]
```diff
@@ -577,6 +577,7 @@ def TAGLINES():
         "Conches for the xonsh god!",
         "Python-powered, cross-platform, Unix-gazing shell",
         "Tab completion in Alderaan places",
+        "This fix was trickier expected",
```
Comment (gforsyth) [inline xonsh/xonfig.py]: trickier than expected

File: docs/intro.rst (hunk 1 above)
Comment (gforsyth) [reply docs/intro.rst]: trickier *than expected

**Comment by scopatz** [comment]
Ok, I think this is ready for another look!

**Comment by gforsyth** [comment]
Thanks, @scopatz!

**Pull request merged**